File: app/modules/signsight-hand-tracker/android/src/main/java/expo/modules/signsighthandtracker/SignSightHandTrackerModule.kt
package expo.modules.signsighthandtracker

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class SignSightHandTrackerModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("SignSightHandTracker")

    OnCreate {
      val reactContext = appContext.reactContext ?: return@OnCreate
      SignSightHandTrackerFrameProcessorPlugin.registerOnce(reactContext.applicationContext)
    }

    Function("getPluginName") {
      SignSightHandTrackerFrameProcessorPlugin.PLUGIN_NAME
    }

    Function("isSupported") {
      true
    }

    Function("getTrackingCapabilities") {
      mapOf(
        "hands" to true,
        "upperBody" to true,
        "gestureV2" to false
      )
    }
  }
}

File: app/modules/signsight-hand-tracker/ios/SignSightHandTrackerModule.swift
import ExpoModulesCore

public class SignSightHandTrackerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("SignSightHandTracker")

    Function("getPluginName") {
      "signsightDetectHands"
    }

    Function("isSupported") {
      false
    }
  }
}
